kola/tests/docker/docker.go

```diff
@@ -382,3 +388,91 @@ func dockerUserNoCaps(c cluster.TestCluster) {
 		c.Fatalf("reading /root test failed: %q", string(output))
 	}
 }
+
+// testDockerInfo test that docker info's output is as expected.  the expected
+// filesystem may be asserted as one of 'overlay', 'btrfs', 'devicemapper'
+// depending on how the machine was launched.
+func testDockerInfo(expectedFs string, c cluster.TestCluster) {
+	m := c.Machines()[0]
+
+	dockerInfoJson, err := m.SSH(`curl -s --unix-socket /var/run/docker.sock http://docker/v1.24/info`)
+	if err != nil {
+		c.Fatalf("could not get dockerinfo: %v", err)
+	}
+
+	type simplifiedDockerInfo struct {
+		ServerVersion string
+		Driver        string
+		CgroupDriver  string
+		Runtimes      map[string]struct {
+			Path string `json:"path"`
+		}
+		ContainerdCommit struct {
+			ID       string
+			Expected string
+		}
+		RuncCommit struct {
+			ID       string
+			Expected string
+		}
+		SecurityOptions []string
+	}
+
+	info := &simplifiedDockerInfo{}
+	err = json.Unmarshal(dockerInfoJson, &info)
+	if err != nil {
+		c.Fatalf("could not unmarshal dockerInfo %q into known json: %v", string(dockerInfoJson), err)
+	}
+
+	// Canonicalize info as needed
+	sort.Strings(info.SecurityOptions)
+
+	// Because we prefer overlay2/overlay for different docker versions, figure
+	// out the correct driver to be testing for based on our docker version.
+	var expectedOverlayDriver string
+	if strings.HasPrefix(info.ServerVersion, "1.12.") {
+		expectedOverlayDriver = "overlay"
+	} else if strings.HasPrefix(info.ServerVersion, "17.04") {
+		expectedOverlayDriver = "overlay"
+	} else if strings.HasPrefix(info.ServerVersion, "17.") {
+		expectedOverlayDriver = "overlay2"
+	} else {
```

Comment:
Rather than breaking the test next January, I'd suggest defaulting to `overlay2`.  That will produce failures on old Docker versions, which is okay because we'll still notice it.